forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -636,15 +636,30 @@ private Map<String, Set<Hash>> getCommitTitleToCommitsMap(ReadOnlyRepository loc
                 .param("all", "true")
                 .execute()
                 .asArray();
+
+        Map<String, Set<Hash>> tempCommitTitleToCommits = new HashMap<>();
+
         for (var commit : commits) {
             var hash = new Hash(commit.get("id").asString());
             var title = commit.get("title").asString();
-            commitTitleToCommits.computeIfAbsent(title, t -> new LinkedHashSet<>()).add(hash);
+            tempCommitTitleToCommits.computeIfAbsent(title, t -> new LinkedHashSet<>()).add(hash);
```

Comment:
I read the bug description and I think I understand the problem now. The initial creation of the map iterates commits from newest to oldest, while this loop iterates from oldest to newest.

Instead of creating a temporary map, couldn't we just call `LinkedHashSet::addFirst` to get the new commits added at the front in the correct order?

Reply:
Yes, it's better to use `LinkedHashSet::addFirst`